server/src/main/java/server/connection/start/ClientRegistrationListener.java

```diff
@@ -28,7 +28,7 @@ public void run() {
         this.running = true;
         while (isRunning()) {
             try {
-                new StartClientHandler(serverSocket.accept(), userFactory, userManager).start();
+                new ClientRegistrationHandler(serverSocket.accept(), userFactory, userManager).start();
```

Comment:
This is second thread you start. Do you really need another thread right now? You think register new client after pulling new Socket from serverSocket.accept() and then block again on serverSocket.accept()